src/libstd/sys/windows/fs2.rs

```diff
@@ -111,8 +120,31 @@ impl DirEntry {
     }
 
     pub fn path(&self) -> PathBuf {
+        self.root.join(&self.file_name())
+    }
+
+    pub fn file_name(&self) -> OsString {
```

Comment:
Like I pointed out in the RFC (a little lately, sorry for that). If the entry is a directory or something else, is file_name is still a good method name ?